modules/sor/sorV2/lib/pools/weighted/weightedPool.ts

```diff
@@ -64,17 +66,13 @@ export class WeightedPool implements BasePool {
                 poolToken.token.symbol,
                 poolToken.token.name,
             );
+            // TODO: parseFloat with toFixed changes the value, better use parseEther instead
             const balance = parseFloat(poolToken.dynamicData.balance).toFixed(18);
             const tokenAmount = TokenAmount.fromHumanAmount(token, balance as `${number}`);
-
-            poolTokens.push(
-                new WeightedPoolToken(
-                    token,
-                    tokenAmount.amount,
-                    parseEther(poolToken.dynamicData.weight),
-                    poolToken.index,
-                ),
-            );
+            //TODO: Remove this once the weight for V3 Pools is in the same format as V2 Pools
```

Comment:
@franzns - is there an expectation to when this will be fixed?

Reply:
we should fix that before we merge this.

Reply:
We have the weights now as floats in the subgraph and the db. Can you change this?

Reply:
yeap - will do 👍